crates/core/src/rlp/decode.rs

```diff
@@ -313,12 +312,34 @@ impl<T1: RLPDecode, T2: RLPDecode, T3: RLPDecode> RLPDecode for (T1, T2, T3) {
     }
 }
 
+impl<T1: RLPDecode, T2: RLPDecode, T3: RLPDecode, T4: RLPDecode> RLPDecode for (T1, T2, T3, T4) {
```

Comment:
is this one really needed? taking into account that we have the Decoder helper for structs?